content/md/en/docs/main-docs/build/build-process.md

```diff
@@ -1,118 +1,155 @@
 ---
 title: Build process
-description: This article describes what happens under the hood when building and executing a Substrate node.
+description: Describes how a Substrate node is compiled into Rust and WebAssembly binaries and how the binaries are used to execute calls into the Substrate runtime.
 keywords:
 ---
 
-This article describes what happens under the hood when building and executing a Substrate node.
+In [Architecture](/main-docs/fundamentals/architecture), you learned that a Substrate node consists of two main environments—an outer node host environment and a runtime execution environment—and that these environments communicate with each other through runtime API calls and host function calls.
+In this section, you'll learn more about how the Substrate runtime is compiled into a standard Rust executable that is included as part of the node executable and into a WebAssembly (Wasm) binary that is stored on the blockchain.
+After you see the inner-working of how the binaries are compiled, you'll learn more about why there are two binaries, when they are used, and how you can change the execution strategies, if you need to.
 
-Relevant context:
+## Rust runtime
 
-- Runtimes should always be compressed Wasm binaries for on-chain upgrades and relay chain validation capabilities to function correctly.
-- Read about Substrate's [Architecture](/main-docs/fundamentals/architecture) in order to understand how the client interacts with the runtime.
+The Rust version of the runtime is used as a regular Rust crate dependency to the Substrate node executable.
+You can see this by displaying the package dependencies for the `node-template-runtime` package.
+For example:
 
-## Compilation
+1. Open a terminal on a computer where you have the node template installed.
 
-Substrate runtimes are compiled to both native and Wasm.
-The native runtime is used as a regular Rust crate dependency to the Substrate client.
-You can see this when running `cargo tree -i node-template-runtime` from the terminal of a standard Substrate node template:
+1. Change to the root of the node template directory.
 
-```bash
-# Displays all packages that depend on the node-template-runtime package.
-node-template-runtime devhub/latest (...)
-└── node-template devhub/latest (...)
-```
+1. Display an inverted dependency graph by running the following command:
 
-But how is the Wasm runtime created?
-What really happens when `cargo build --release` is executed in the directory of a Substrate node project?
+    ```bash
+    cargo tree -i node-template-runtime
+    ```
```

Comment:
This whole section is very small, and makes no sense to me to understand anything.

Look at some rust dependencies and understand what?